react/src/components/Widget/Widget.tsx

```diff
@@ -263,46 +262,50 @@ export const Widget: React.FC<WidgetProps> = props => {
     }
     url = prefixedAddress + (query.length ? `?${query.join('&')}` : '');
 
+    let addressType: currency = 'AUD';
+
+    if (isValidCashAddress(address)) {
+      addressType = 'BCH';
+      setWidgetButtonText(`Send with BCH wallet`);
+    } else if (isValidXecAddress(address)) {
+      addressType = 'XEC';
+      setWidgetButtonText(`Send with XEC wallet`);
+    }
+
     if (currencyObj && hasPrice) {
-      const bchAmount = getCurrencyObject(
-        currencyObj.float / (price! / 100),
-        'BCH',
-      );
-
-      if (isValidCashAddress(address)) {
-        setText(
-          `Send ${currencyObj.string} ${currencyObj.currency} = ${bchAmount.BCHstring} BCH`,
-        );
-        setWidgetButtonText(
-          `Send ${currencyObj.string} ${currencyObj.currency} = ${bchAmount.BCHstring} BCH`,
-        );
-      } else if (isValidXecAddress(address)) {
-        setText(
-          `Send ${currencyObj.string} ${currencyObj.currency} = ${bchAmount.BCHstring} XEC`,
-        );
-        setWidgetButtonText(
-          `Send ${currencyObj.string} ${currencyObj.currency} = ${bchAmount.BCHstring} XEC`,
-        );
+      const bchAmount = price
+        ? getCurrencyObject(currencyObj.float / price, addressType)
+        : null;
+
+      if (bchAmount) {
+        if (isValidCashAddress(address)) {
+          setText(
+            `Send ${currencyObj.string} ${currencyObj.currency} = ${bchAmount.string} BCH`,
+          );
+        } else if (isValidXecAddress(address)) {
+          setText(
+            `Send ${currencyObj.string} ${currencyObj.currency} = ${bchAmount.string} XEC`,
+          );
+        }
+        query.push(`amount=${bchAmount.float}`);
       }
-      query.push(`amount=${bchAmount.float}`);
+
       url = prefixedAddress + (query.length ? `?${query.join('&')}` : '');
       setUrl(url);
     } else {
       const notZeroValue: boolean =
-        currencyObj?.satoshis !== undefined && currencyObj.satoshis > 0;
-      if (!isFiat && currencyObj && notZeroValue) {
+        currencyObj?.float !== undefined && currencyObj.float > 0;
+      if (!isFiat(currency) && currencyObj && notZeroValue) {
         const bchType: string = currencyObj.currency;
         setText(`Send ${currencyObj.string} ${bchType}`);
-        query.push(`amount=${currencyObj.BCHstring}`);
+        query.push(`amount=${currencyObj.string}`);
         url = prefixedAddress + (query.length ? `?${query.join('&')}` : '');
         setUrl(url);
       } else {
         if (isValidCashAddress(address)) {
           setText(`Send any amount of BCH`);
-          setWidgetButtonText(`Send any amount of BCH`);
         } else if (isValidXecAddress(address)) {
           setText(`Send any amount of XEC`);
-          setWidgetButtonText(`Send any amount of XEC`);
         }
```

Comment:
Looks like you could leverage your validCurrencyType function here again to simplify things

```js
const addressCurrency = validCurrencyType(address);
setText(`Send any amount of ${addressCurrency}`);
setWidgetButtonText(`Send any amount of ${addressCurrency}`);
```

Reply:
done